### ISSUE TITLE
[GraphQL] Support query arguments in trace metadata

### ISSUE THREAD

**Issue by xzyfer** [DataDog/dd-trace-rb#777]
In dd-trace-js the GraphQL plugin accepts a whitelist of variable names that should be added as `execute.graphql` span metadata for query operations.

```js
tracer.use('graphql', {
  analytics: true,
  variables: function(provided_variables) {
    return provided_variables;
  }
});
```
```js
tracer.use('graphql', {
  analytics: true,
  variables: ['variable_name_1', 'variable_name_2']
});
```

See https://github.com/DataDog/dd-trace-js/pull/237

**Comment by delner** [comment]
Can you expand on the use case for the use of `variables` a bit? When would someone use this feature? And how would it work?

**Comment by xzyfer** [comment]
Datadog allows for creating facets within their trace analytics. We use this to differentiate response time percentiles for a query. 

For example the query above `withCategorySet` has a variable `categoryId`. The category is a reference to a set. The size of each set is fixed anywhere between 1 and 500 depending on the input categoryId. As a result the response percentiles for this query are worse than averages (essentially useless). With the JS tracer we can facet this query by the value of the categoryId query which allows us to get analytics on similarly size sets.

### Median across all categories

![image](https://user-images.githubusercontent.com/579928/60575587-2d8c6200-9dbf-11e9-8958-634c5e9106dc.png)

### Median across small sized categories

![image](https://user-images.githubusercontent.com/579928/60575651-48f76d00-9dbf-11e9-8ac1-649c3f068ba5.png)

### Median across medium sized categories

![image](https://user-images.githubusercontent.com/579928/60575696-5dd40080-9dbf-11e9-9766-27129a3f61f5.png)

### Median across large sized categories

![image](https://user-images.githubusercontent.com/579928/60575738-6c221c80-9dbf-11e9-8f0d-c916550873ff.png)

**Comment by delner** [comment]
Ahh, okay I see. So if I'm understanding correctly, the goal is to allow whitelisted variables into tags, which can then be converted to a facet with trace search & analytics?

**Comment by xzyfer** [comment]
Correct. A subtle but important feature is that the whitelist support a function signature. This is neccessy because many GraphQL client libraries will reqry variable names in the query AST i.e. apollo will prefix variable names resulting in something like `_v0_categoryId`.

We need a way to express that `categoryId` and `_v0_categoryId` and `_v1_cateforyId` should result in the same `categoryId` tag.
